glossary.md

```diff
@@ -13,52 +13,73 @@ gh: specifically for github
+**commit**: Saves the staged changes to your LOCAL repository. It creates a *snapshot* of the repository in it's current statement
 
-**push**: Sends your local committed changes to the remote repository.
+**push**: Sends your local committed changes to the remote repository. The staging area is now empty.
 
 
 | command    | use |
 | -------- | ------- |
-| git remote add [alias] [url]| add a git url as an alias. The alias is usually 'origin'    |
-| git fetch [alias] |fetch down all the branches from the git remote. Retreive updates from a remote repo but doesn't merge them into your current working branch. You can review changes before integrating them     |
-| git merge [alias]/[branch] | combines changes from one branch to another branch |
-| git pull | takes all commits from a remote repository and integrates them into yur current branch | 
-| git push [alias][branch] | sends local branch commits to remote repository branch | 
-| git commit -m "[message]" | commits your current content as a new commit with a message to add context |
+| git remote add [alias] [url]| This command lets you connect your personal code to the shared repository. The 'git remote add' part tells Git you want to add a new connection to the remote repository with the nickname '[alias]'. The URL is the location of the shared repository. |
+| git fetch [alias] |Let's say your friend updated the repo and you want to review the changes before putting them in your code. You use 'git fetch' to get the latest updates from the remote repo and the '[alias]' is the same nickname from when you added the repository in the previous command. However, git fetch DOES NOT add the changes to your code automatically.     |
+| git merge [alias]/[branch] | Let's say you reviewed their code and you're ready to integrate that code to your code. Git merge does just that-- it adds the changes to your code.  |
+| git pull | This basically does the work of git fetch and git merge in one command. It's when you know fosho that your friends code is good and you want to add it to your repository. | 
+|git add [filename] | This command adds the specific changed file you're referencing to in [filename] to the staging area. You're adding your new notes/changes into the tempoary section of the binder. |
+|git add . | This commands all the files that you changed to the staging area. The '.' is what makes it ALL the files you've changed | 
+| git push [alias][branch] | Now you've coded a function and you want to add that to the online repo. You use git push to add your code. Sometimes, you have to make a PR (pull request).  | 
+| git commit -m "[message]" | commits your current content as a new commit with a message to add context. Whenever you commmit, it's a good idea to add a message for context to explain what you're commiting |
+|git head| Similar to a bookmark telling you which note you're on, git head tells you which commit you're working on. |
```

Comment:
this is a good explanation for a difficult concept, but I think we can do better? I think the term bookmark implies spatial marking where HEAD is more of a temporal/branch type notion. It's more like the revision of a textbook than a bookmark within that book.